app/jobs/dailyreport_job.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+class DailyreportJob < ApplicationJob
+  queue_as :default
+
+  def perform
+    report_content = daily_report_job
+    send_report_to_slack(report_content)
+  end
+
+  def daily_report_job
+    users = User.where(created_at: Date.yesterday.beginning_of_day..Date.yesterday.end_of_day).count
+    posts = Micropost.where(created_at: Date.yesterday.beginning_of_day..Date.yesterday.end_of_day, parent_id: nil).count
```

Comment:
Use scope in model for `parent_id: nil`, `created_at: Date.yesterday.beginning_of_day..Date.yesterday.end_of_day`